c/dynalloc.c

```diff
@@ -959,7 +959,60 @@ int unallocDataset(DynallocInputParms *inputParms, int *reasonCode) {
 
 }
 
+int freeDataset(DynallocInputParms *inputParms, int *reasonCode) {
```

Comment:
This is a bad name. It unallocates a dataset and then because ```DISP_DELETE``` has been used, the dataset gets removed. It'd be better to add a flag value to the ```dynallocUnallocDatasetByDDName``` function.